torch/utils/checkpoint.py

```diff
@@ -275,3 +287,85 @@ def forward(input):
         input = checkpoint(run_function(start, end, functions), input,
                            preserve_rng_state=preserve)
     return run_function(end + 1, len(functions) - 1, functions)(input)
+
+
+class Checkpoint(torch.nn.Module):
```

Comment:
Implementation is entirely from https://github.com/pytorch/pytorch/pull/62964

Reply:
Why is this an nn.Module? This should be just a function no?

Reply:
It is indeed an nn.Module but it is basically called as a function in L223. Do you mean this should be a torch.autograd.Function?

The implementation is directly from https://github.com/pytorch/pytorch/pull/62964/files, do we want to refactor this to be a function or stick with the original implementation?

Reply:
I meant it could be a plain old python function.
This is not going to be public API anyways.